finetune/base_models/tcn/featurizer.py

```diff
@@ -0,0 +1,138 @@
+        
+        class TemporalBlock(object):
+            def __init__(self, n_filters, kernel_size, dilation_rate, rate, scope, layer_num):
+                self.rate=rate
+                self.n_filters = n_filters
+                self.conv1 = tf.keras.layers.Conv1D(
+                    filters=n_filters,
+                    kernel_size=kernel_size,
+                    padding="same",
+                    activation=tf.nn.relu,
+                    dilation_rate=dilation_rate,
+                    kernel_initializer=tf.initializers.glorot_normal,
+                    name="conv1_" + str(layer_num),
+                )
+                self.conv2 = tf.keras.layers.Conv1D(
+                    filters=n_filters,
+                    kernel_size=kernel_size,
+                    padding="same",
+                    dilation_rate=dilation_rate,
+                    activation=tf.nn.relu,
+                    kernel_initializer=tf.initializers.glorot_normal,
+                    name="conv1_" + str(layer_num),
+                )
+                self.downsample = tf.keras.layers.Conv1D(filters=n_filters, kernel_size=1, padding="same")
+
+            def __call__(self, X):
+                conv1_out = self.conv1(X)
+                conv1_dropout = tf.nn.dropout(conv1_out, rate=self.rate)
+                conv2_out = self.conv2(conv1_dropout)
+                conv2_dropout = tf.nn.dropout(conv2_out, rate=self.rate)
+
+                # residuals
+                if X.get_shape()[-1] == self.n_filters:
+                    output = conv2_dropout + X
+                else:
+                    downsampled_out = self.downsample(X)
+                    output = conv2_dropout + downsampled_out
+
+                return output
+
+        with tf.variable_scope("crf"):
+            representation = h
+            for layer_num in range(config.n_layer):
+                representation = TemporalBlock(
+                    n_filters=config.n_filter,
+                    kernel_size=config.kernel_size,
+                    rate=config.resid_p_drop if train else 0,
+                    dilation_rate=2 ** layer_num,
+                    scope="Temporal{}".format(layer_num),
+                    layer_num=layer_num
+                )(representation)
+
+        seq_feats = tf.reshape(representation, shape=[-1, config.max_length, config.n_filter])
+
+        # mask out the values past the classify token before performing pooling
+        pool_idx = tf.cast(
+            tf.argmax(tf.cast(tf.equal(X[:, :, 0], clf_token), tf.float32), 1),
+            tf.int32,
+        )
+
+        # mask is past the classify token (i.e. make those results extremely negative)
+        mask = tf.expand_dims(
+            1.0
+            - tf.sequence_mask(
+                pool_idx, maxlen=tf.shape(representation)[1], dtype=tf.float32
+            ),
+            -1,
+        )
+        pool = tf.reduce_max(representation + mask * -1e9, 1)
```

Comment:
The masking and pooling is the same as for the textcnn. Could you pull these out to a shared function?